src/backend/joanie/edx_imports/tasks/certificates.py

```diff
@@ -0,0 +1,206 @@
+    return report
+
+
+def import_certificates_batch(start, stop, dry_run=False):
+    """Batch import certificates from Open edX certificates_generatedcertificate"""
+    db = OpenEdxDB()
+    report = {
+        "certificates": {
+            "created": 0,
+            "updated": 0,
+            "errors": 0,
+        }
+    }
+    hashids = Hashids(salt=settings.EDX_SECRET)
+    certificates = db.get_certificates(start, stop)
+    certificates_to_create = []
+    certificates_to_update = []
+
+    for edx_certificate in certificates:
+        try:
+            enrollment = models.Enrollment.objects.get(
+                user__username=edx_certificate.user.username,
+                course_run__resource_link__icontains=edx_certificate.course_id,
+            )
+        except models.Enrollment.DoesNotExist:
+            report["certificates"]["errors"] += 1
+            logger.error(
+                "No Enrollment found for %s %s",
+                edx_certificate.user.username,
+                edx_certificate.course_id,
+            )
+            continue
+
+        organization_code, signatory = edx_mongodb.get_enrollment(
+            edx_certificate.course_id
+        )
+
+        if not organization_code:
+            report["certificates"]["errors"] += 1
+            logger.error(
+                "No organization found in mongodb for %s", edx_certificate.course_id
+            )
+            continue
+
+        try:
+            organization = models.Organization.objects.get(code=organization_code)
+        except models.Organization.DoesNotExist:
+            report["certificates"]["errors"] += 1
+            logger.error("No organization found for %s", organization_code)
+            continue
+
+        verification_hash = hashids.encode(edx_certificate.id)
+
+        signature = None
+        if signatory:
+            signature_image_path = signatory.get("signature_image_path")
+            if signature_image_path.startswith("/"):
+                signature_image_path = signature_image_path[1:]
+            signature_path = download_and_store(signature_image_path)
+            signature = image_to_base64(default_storage.path(signature_path))
+
+        certificate_context = {
+            "signatory": signatory,
+            "verification_hash": verification_hash,
+        }
+
+        title_object = enrollment.course_run.course
+
+        for language, _ in settings.LANGUAGES:
+            certificate_context[language] = {
+                "course": {
+                    "name": title_object.safe_translation_getter(
+                        "title", language_code=language
+                    ),
+                },
+                "organizations": [
+                    {
+                        "name": organization.safe_translation_getter(
+                            "title", language_code=language
+                        ),
+                        "representative": organization.representative,
```

Comment:
If I remember well, this information should comes from the certificate itself ?
Furthermore, for imported universities, `representative` field will be empty.